src/app/team/documents/proofs/DocsProofs.jsx

```diff
@@ -2,45 +2,45 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
 import { compose, withProps } from 'recompose';
-
-import { getDocuments, getFirstProof } from '../../../../selectors/document';
+import { getDocuments, getRequiredDocuments } from '../../../../selectors/document';
 import { getServiceId } from '../../../../selectors/service';
 import { selectLocationError } from '../../../../selectors/location';
-
 import * as actions from '../../../../actions';
-import { Form, FormEdit, FormView } from '../../../../components/form';
-
-const EditComponent = compose(withProps({
-  headerText: 'What proofs are required to use this service?',
-  placeholderText: 'e.g. Proof of identity, proof of address',
-}))(props => <FormEdit {...props} />);
-
-const ViewComponent = compose(withProps({
-  topText: 'PROOFS REQUIRED',
-}))(props => <FormView {...props} />);
+import { Form } from '../../../../components/form';
+import Edit from './Edit';
+import View from './View';
 
 const FormComponent = compose(withProps({
-  ViewComponent,
-  EditComponent,
+  ViewComponent: View,
+  EditComponent: Edit,
 }))(props => <Form {...props} />);
 
 const mapStateToProps = (state, ownProps) => ({
   resourceData: getDocuments(state, ownProps),
-  value: getFirstProof(state, ownProps),
+  value: getRequiredDocuments(state, ownProps),
   id: getServiceId(ownProps),
   resourceLoadError: selectLocationError(state, ownProps),
 });
 
 const mapDispatchToProps = (dispatch, ownProps) => ({
   fetchResourceData: bindActionCreators(actions.getLocation, dispatch),
-  updateValue: (value, serviceId, metaDataSection, fieldName) =>
-    dispatch(actions.updateService({
+  updateValue: (proofs, serviceId, metaDataSection, fieldName) => {
+    if (proofs.every(p => (typeof p !== 'string'))) {
```

Comment:
So looks like this is checking if the proofs array has at least 1 real value in it.
That makes sense, but then what will happen with any values that are _not_ strings? If they're still sent to the action without any filtering, will they end up deleting existing documents, or...?

Reply:
I think it was meant to be a null check, but is not required

Reply:
Oh, OK. I hope that was indeed the only edge case this was meant to cover.